src/collectors/common-contexts/system.interrupts.h

```diff
@@ -0,0 +1,61 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+
+#ifndef NETDATA_SYSTEM_INTERRUPTS_H
+#define NETDATA_SYSTEM_INTERRUPTS_H
+
+#include "common-contexts.h"
+
+#define _system_interrupt_chart() \
+   rrdset_create_localhost( \
+        "system" \
+        , "intr" \
+        , NULL \
+        , "interrupts" \
+        , NULL \
+        , "CPU Interrupts" \
+        , "interrupts/s" \
+        , _COMMON_PLUGIN_NAME \
+        , _COMMON_PLUGIN_MODULE_NAME \
+        , NETDATA_CHART_PRIO_SYSTEM_INTR \
+        , update_every \
+        , RRDSET_TYPE_LINE \
+        )
+
+#ifdef OS_WINDOWS
+static inline void common_interrupts(COUNTER_DATA *interrupts, int update_every) {
+    static RRDSET *st_intr = NULL;
+    static RRDDIM *rd_interrupts = NULL;
+
+    if(unlikely(!st_intr)) {
+        st_intr = _system_interrupt_chart();
+
+        rrdset_flag_set(st_intr, RRDSET_FLAG_DETAIL);
+
+        rd_interrupts = rrddim_add(, , NULL, 1, 1, RRD_ALGORITHM_INCREMENTAL);
+        rd_interrupts = perflib_rrddim_add(st_intr, "interrupts", NULL, 1, 1, interrupts);

```

Comment:
Why do we need `perflib_rrddim_add` for system interrupts? What problem did you have with prev version of the `common_interrupts`?

Reply:
It is because we are debugging issues like this:

![Screenshot_20240528_140817](https://github.com/netdata/netdata/assets/49162938/2024a1f3-1a0b-4bb0-a1e8-63bae541e1a7)

Reply:
If you are debugging something convert this PR to Draft.

Reply:
And "Interrupts/sec" (so as other metrics from Processor) is per core.  I may have missed that, but I think your code doesn't consider it.

Reply:
> And "Interrupts/sec" (so as other metrics from Processor) is per core. I may have missed that, but I think your code doesn't consider it.

The charts need to be in the `processor` structure. This will ensure that it will be handled per core. Check this

```
  p = dictionary_set(processors, windows_shared_buffer, NULL, sizeof(*p));
  is_total = false;
  cpu = str2i(windows_shared_buffer);
  snprintfz(windows_shared_buffer, sizeof(windows_shared_buffer), "cpu%d", cpu);

  if(cpu + 1 > cores_found)
     cores_found = cpu + 1;
```

The windows_shared_buffer will have the cpu core and (thus gets a new processor structure per core). Note that in some counters (I do not know about this one) the windows_shared_buffer will return two numbers separated by (NUMA, CORE) so parsing the cpu as it is now will cause issues.